rest_sensors.cpp

```diff
@@ -1354,19 +1354,99 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
                         }
                     }
                 }
-                else if ((rid.suffix == RConfigLocked) && (sensor->modelId().startsWith(QLatin1String("kud7u2l")) ||
-                                                           sensor->modelId().startsWith(QLatin1String("GbxAXL2")) ||
-                                                           sensor->modelId().startsWith(QLatin1String("TS0601"))))
+                else if (rid.suffix == RConfigLocked)
                 {
-                    QByteArray data = QByteArray("\x00", 1);
-                    if (map[pi.key()].toBool())
+                    if (map[pi.key()].type() == QVariant::Bool)
+                    {
+                        if (sensor->modelId().startsWith(QLatin1String("kud7u2l")) || sensor->modelId().startsWith(QLatin1String("GbxAXL2")) ||
+                            sensor->modelId().startsWith(QLatin1String("TS0601")))
+                        {
+                            QByteArray data = QByteArray("\x00", 1);
+                            if (map[pi.key()].toBool())
+                            {
+                                data = QByteArray("\x01", 1);
+                            }
+
+                            if (SendTuyaRequest(task, TaskThermostat , 0x0107, data))
+                            {
+                                updated = true;
+                            }
+                        }
+                        else if (sensor->modelId() == QLatin1String("eTRV0100") || sensor->modelId() == QLatin1String("TRV001"))
+                        {
+                            quint32 data = map[pi.key()].toUInt(&ok);
+
+                            if (addTaskThermostatUiConfigurationReadWriteAttribute(task, deCONZ::ZclWriteAttributesId, 0x0001, deCONZ::Zcl8BitEnum, data))
+                            {
+                                updated = true;
+                            }
+                            else
+                            {
+                                rsp.list.append(errorToMap(ERR_ACTION_ERROR, QString("/sensors/%1/config/locked"), QString("Could not set attribute")));
+                                rsp.httpStatus = HttpStatusBadRequest;
+                                return REQ_READY_SEND;
+                            }
+                        }
+                    }
+                    else
                     {
-                        data = QByteArray("\x01", 1);
+                        rsp.list.append(errorToMap(ERR_INVALID_VALUE, QString("/sensors/%1/config/locked"), QString("Invalid attribute value")));
```

Comment:
`QString("/sensors/%1/config/locked")`

Is the `.arg(...)` missing for %1 argument?

Reply:
Amended.